python/lsst/sims/alertsim/catalogs/dia_transformations.py

```diff
@@ -0,0 +1,151 @@
+""" dia_transformations """
+from lsst.sims.photUtils import Sed  # for converting magnitudes into fluxes
+from random_utils import array_to_dict
+import numpy as np
+import re
+import random
+
+"""
+A module which groups methods for transformation of catsim values 
+to diaSource attributes as stated in DPDD.
+These methods needed to be separated from catalog classes 
+because ad-hoc usage is required.
+"""
+
+def midPointTai(tai):
+    """
+    Return mid point of exposure by taking OpSim start-of-exposure time
+    and adding 17 seconds (15 seconds for first exposure; 1 second for
+    shutter close; one second for shutter open).  Ignore the fact that,
+    as DPDD states, midpoint will vary for different objects based on
+    their position relative to the shutter motion.
+    """
+    return tai+17.0/86400.0
+
+
+def ccdVisitId(obsHistID, chipNum):
+    """
+    Previous solution:
+    Return chipNum*10^7 + obsHistID (obsHistID should never be more than 3 million)
+    This was no good as 0220 was same as 2200
+    Now:
+    Return obsHistID*10^4 + chipNum
+    """
+    return obsHistID*10000+chipNum
+
+def chipNum(chip_name):
+    """
+    Concatenate the digits in 'R:i,j S:m,n' to make the chip number ijmn
+    """
+    """
+    for idx, val in enumerate(chip_name):
+        if val is None: chip_name[idx] = '0'
+        
+    chip_arr = np.array([int(''.join(re.findall(r'\d+', name))) for name in chip_name])
+    return chip_arr
+    """
+    return np.array([int(''.join(re.findall(r'\d+', name))) if name is not None else 0
+                     for name in chip_name])
+
+def diaSourceId(uniqueId, obsHistID):
+    """
+    A unique identifier for each DIASource (this needs to be unique for
+    each apparition of a given object)
+
+    Take uniqueID, multiply by 10^7 and add obsHistID from self.obs_metadata
+    (obsHistID should only go up to about 3 million)
+    """
+    return uniqueId*10000000 + obsHistID
+
+def fluxFromMag(mag):
+    """
+    Use Sed to convert mag into flux
+    """
+    ss = Sed()
+    return ss.fluxFromMag(mag)
+
+def addEpsilon(some_value):
```

Comment:
It does not look like this method is used.  You can delete it.

Reply:
I think I'll need it, commented out and moved to bottom for now.